mypy errors in history.py

```
$ make mypy
python3 -m mypy arkouda
arkouda/history.py:118: error: Incompatible return value type (got "list[str | tuple[str, str | None]]", expected "list[str]")  [return-value]
arkouda/history.py:121: error: Argument 1 to "_filter_arkouda_command" of "HistoryRetriever" has incompatible type "str | tuple[str, str | None]"; expected "str"  [arg-type]
arkouda/history.py:124: error: Incompatible return value type (got "list[str | tuple[str, str | None]]", expected "list[str]")  [return-value]
arkouda/numpy/util.py:636: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/numpy/segarray.py:71: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 3 errors in 1 file (checked 94 source files)
make: *** [Makefile:705: mypy] Error 1

```

--- a/arkouda/history.py
+++ b/arkouda/history.py
@@ -15,6 +15,7 @@ class HistoryRetriever:
         Returns command string if the filter string is in the command and the
         command is not generate_history. Otherwise, returns None
         """
+
         return command if (filter_string in command and "generate_history" not in command) else None
 
     def retrieve(
@@ -63,6 +64,17 @@ class ShellHistoryRetriever(HistoryRetriever):
         -------
         List[str]
             A list of commands from the Python shell, Jupyter notebook, or IPython notebook
+
+        Examples
+        --------
+        >>> from arkouda.history import ShellHistoryRetriever
+        >>> import readline
+        >>> h = ShellHistoryRetriever()
+        >>> readline.clear_history()
+        >>> 1 + 2
+        3
+        >>> h.retrieve()
+        [' 1 + 2', 'h.retrieve()']
         """
         length_of_history = readline.get_current_history_length()
         num_to_return = num_commands if num_commands else length_of_history
@@ -103,6 +115,16 @@ class NotebookHistoryRetriever(HistoryAccessor, HistoryRetriever):
         -------
         List[str]
             A list of commands from the Python shell, Jupyter notebook, or IPython notebook
+
+        Examples
+        --------
+        >>> from arkouda.history import NotebookHistoryRetriever
+        >>> h = NotebookHistoryRetriever()
+        >>> 1+2
+        >>> 4*6
+        >>> 2**3
+        >>> h.retrieve(num_commands=3)
+        ['1+2', '4*6', '2**3']
         """
         raw = True  # HistoryAccessor _run_sql method parameter
         output = False  # HistoryAccessor _run_sql method parameter
@@ -114,11 +136,8 @@ class NotebookHistoryRetriever(HistoryAccessor, HistoryRetriever):
         else:
             cur = self._run_sql("ORDER BY session DESC, line DESC", (), raw=raw, output=output)
 
+        ret = [cmd[2] for cmd in reversed(list(cur)) if isinstance(cmd[2], str)]
         if command_filter:
-            return [
-                cmd[2]
-                for cmd in reversed(list(cur))
-                if self._filter_arkouda_command(cmd[2], command_filter)
-            ][-num_to_return:]
-        else:
-            return [cmd[2] for cmd in reversed(list(cur))][-num_to_return:]
+            ret = [cmd for cmd in ret if self._filter_arkouda_command(cmd, command_filter)]
+
+        return ret[-num_to_return:]

--- a/setup.cfg
+++ b/setup.cfg
@@ -26,3 +26,4 @@ exclude =
     pydoc/_ext/generic_linkcode_resolve_for_sphinx.py
     converter/csv2hdf.py
     ./build
+    ./.git

